### PR TITLE
Support more languages (git branch)

### PR DESCRIPTION
**Pull request by mattseddon**
Follow up from #4137

In response to [this](https://discord.com/channels/485586884165107732/1120132608962084914/1120391881940009123).

### CODE REVIEW

File: extension/src/cli/git/reader.test.ts [hunk 1]
```diff
@@ -54,6 +54,7 @@ describe('GitReader', () => {
       expect(mockedCreateProcess).toHaveBeenCalledWith({
         args: ['branch'],
         cwd,
+        env: { LANG: 'en_US.UTF-8' },
```
Comment (mattseddon) [inline extension/src/cli/git/reader.test.ts:57]: [F] Rather than try to replace all possible languages' version of `HEAD detached at` it seems smarter to always ask for the English version and convert that.

File: extension/src/cli/git/reader.ts [hunk 2]
```diff
@@ -151,4 +161,11 @@ export class GitReader extends GitCli {
   private getUris(repositoryRoot: string, relativePaths: string[]) {
     return relativePaths.map(path => resolve(repositoryRoot, path))
   }
+
+  private cleanUpBranchName(branch: string) {
```
Comment (mattseddon) [inline extension/src/cli/git/reader.ts:165]: [F] This is not a very general function so I moved it out of `stdout`.

File: extension/src/cli/git/reader.test.ts (hunk 1 above)
Comment (julieg18) [reply extension/src/cli/git/reader.test.ts:57]: Makes sense! Though I wonder how other extensions handle things like this 🤔

**Review by julieg18 — APPROVED** [review]
Looks good!

File: extension/src/cli/git/reader.test.ts (hunk 1 above)
Comment (mattseddon) [reply extension/src/cli/git/reader.test.ts:57]: Gitlens:

image

image

image

tl;dr - if the branch name has whitespace mangle the sha into `(${short sha}...)`

Comment (mattseddon) [reply extension/src/cli/git/reader.test.ts:57]: We can overhaul when we get to #4111
